testsuite/src/test/java/org/ironjacamar/core/connectionmanager/pool/dflt/CapacityNoCredentialsTestCase.java

```diff
@@ -0,0 +1,397 @@
+            .extract(mcp, "listeners");
+      assertNotNull(listeners);
+      assertEquals(1, TestUtils.collectionSizeTenSecTimeout(listeners, 1));
+      //assertTrue(TestUtils.isCorrectCollectionSizeTenSecTimeout(listeners, 10));
+
+      UnifiedSecurityConnection c = noTxCf.getConnection();
+      assertNotNull(c);
+      assertNull(c.getUserName());
+      assertNull(c.getPassword());
+
+      UnifiedSecurityConnection c1 = noTxCf.getConnection();
+      assertNotNull(c1);
+      assertNull(c1.getUserName());
+      assertNull(c1.getPassword());
+
+      assertNotEquals(c, c1);
+      assertEquals(20, TestUtils.collectionSizeTenSecTimeout(listeners, 20));
+
+      assertEquals(1, mcps.size());
+      ((DefaultManagedConnectionPool) mcp).setLastIdleCheck(Long.MIN_VALUE);
+      mcp.removeIdleConnections();
+      assertEquals(2, TestUtils.collectionSizeTenSecTimeout(listeners, 2));
+
+
+      c1.close();
+
+      c.close();
+
+      // We cheat and shutdown the pool to clear out mcps
+      defaultPool.shutdown();
+   }
+
+   /**
+    * Deployment test w/o Tx
+    *
+    * @throws Throwable In case of an error
+    */
+   @Test
+   public void testFIFO() throws Throwable
+   {
+      assertNotNull(noTxCf);
+      assertNotNull(dr);
+
+      assertEquals(2, dr.getDeployments().size());
+
+      org.ironjacamar.core.api.deploymentrepository.Deployment d = dr
+            .findByJndi("java:/eis/CapacityNoTxConnectionFactory");
+      assertNotNull(d);
+
+      org.ironjacamar.core.api.deploymentrepository.ConnectionFactory dcf = d.getConnectionFactories().iterator()
+            .next();
+      assertNotNull(dcf);
+
+      org.ironjacamar.core.api.deploymentrepository.Pool p = dcf.getPool();
+      assertNotNull(p);
+
+      DefaultPool defaultPool = (DefaultPool) p.getPool();
+
+      ConcurrentHashMap<Credential, ManagedConnectionPool> mcps =
+            (ConcurrentHashMap<Credential, ManagedConnectionPool>) TestUtils.extract(defaultPool, "pools");
+      assertNotNull(mcps);
+
+      assertEquals(1, mcps.size());
+
+      ManagedConnectionPool mcp = mcps.values().iterator().next();
+      assertNotNull(mcp);
+
+      ConcurrentLinkedDeque<ConnectionListener> listeners = (ConcurrentLinkedDeque<ConnectionListener>) TestUtils
+            .extract(mcp, "listeners");
+      assertNotNull(listeners);
+      assertTrue(TestUtils.isCorrectCollectionSizeTenSecTimeout(listeners, 1));
+
+      UnifiedSecurityConnection c = noTxCf.getConnection();
+      assertNotNull(c);
+      assertNull(c.getUserName());
+      assertNull(c.getPassword());
+      UnifiedSecurityConnection c1 = noTxCf.getConnection();
+      assertNotNull(c1);
+      assertNull(c1.getUserName());
+      assertNull(c1.getPassword());
+
+      assertEquals(1, mcps.size());
+      assertEquals(20, TestUtils.collectionSizeTenSecTimeout(listeners, 20));
+
+      c.close();
+      c1.close();
+      assertEquals(noTxCf.getConnection().getListeners().get(0), listeners.getFirst());
```

Comment:
I think it is better to get an identity of the associated ConnectionEventListener through the UnifiedSecurityConnection interface, and check against "listeners". E.g. not expose the ConnectionEventListener directly - same category as ManagedConnection